src/com/facebook/buck/android/DummyRDotJava.java

```diff
@@ -119,9 +123,30 @@ public DummyRDotJava(
           androidResourceDeps,
           rDotJavaSrcFolder,
           forceFinalResourceIds,
-          unionPackage);
+          unionPackage,
+          Optional.<String>absent());
       steps.add(mergeStep);
-      javaSourceFilePaths = mergeStep.getRDotJavaFiles();
+
+      MergeAndroidResourcesStep mergeFinalRStep = null;
+      if (finalRName.isPresent()) {
+        mergeFinalRStep = MergeAndroidResourcesStep.createStepForDummyRDotJava(
+            getProjectFilesystem(),
+            getResolver(),
+            androidResourceDeps,
+            rDotJavaSrcFolder,
+            true,
+            unionPackage,
+            finalRName);
+        steps.add(mergeFinalRStep);
+      }
+
+      if (mergeFinalRStep != null) {
```

Comment:
Combine this into the previous conditional?  And maybe then invert the condition so the shorter case ends up on top?